### ISSUE TITLE
Arrows briefly left hanging on transition to a smaller menu

### ISSUE THREAD

**Issue by cmt218** [saket/cascade#24]
Hi! Just a small UI issue my teammate @burntcookie90 and I were discussing is that during menu transitions sometimes the arrow icon is briefly left hanging. I didn't notice it at all for a while, but then one of our team members pointed it out and now its hard to unsee!

We noticed it in a POC we were doing, but it can actually be seen slightly in the demo gif:

![demo](https://user-images.githubusercontent.com/16960319/106031886-cb303c00-609d-11eb-8078-54e8bc0c6e5d.gif)

Relevant frames (sorry for uneven height):

![Screen Shot 2021-01-27 at 12 45 25 PM](https://user-images.githubusercontent.com/16960319/106032064-fe72cb00-609d-11eb-8e2c-f968a6934f13.png)![Screen Shot 2021-01-27 at 12 45 33 PM](https://user-images.githubusercontent.com/16960319/106032077-03d01580-609e-11eb-9577-404bb85fde20.png)![Screen Shot 2021-01-27 at 12 45 43 PM](https://user-images.githubusercontent.com/16960319/106032108-0e8aaa80-609e-11eb-8306-af446699e3fd.png)![Screen Shot 2021-01-27 at 12 45 51 PM](https://user-images.githubusercontent.com/16960319/106032147-1b0f0300-609e-11eb-846d-a4a1c97b9c62.png)![Screen Shot 2021-01-27 at 12 45 59 PM](https://user-images.githubusercontent.com/16960319/106032161-1f3b2080-609e-11eb-926f-25a45e08ca56.png)![Screen Shot 2021-01-27 at 12 46 07 PM](https://user-images.githubusercontent.com/16960319/106032189-295d1f00-609e-11eb-92bd-f39fec395224.png)![Screen Shot 2021-01-27 at 12 46 15 PM](https://user-images.githubusercontent.com/16960319/106032214-2eba6980-609e-11eb-801c-b3f77addafc6.png)![Screen Shot 2021-01-27 at 12 46 23 PM](https://user-images.githubusercontent.com/16960319/106032227-337f1d80-609e-11eb-88f1-d2e54ba325b7.png)![Screen Shot 2021-01-27 at 12 46 30 PM](https://user-images.githubusercontent.com/16960319/106032250-3e39b280-609e-11eb-814c-a26e6ea5cbd3.png)

Seems to happen when navigating to a shorter menu and the arrows are below the incoming menu items.

Thanks for the great library!

**Comment by saket** [comment]
Interesting. I'm trying to watch a slowed down version of that video and I don't see the arrow left hanging. Is it possible that the low quality of the GIF is giving such an impression? Here's a high quality version of the GIF:

https://user-images.githubusercontent.com/2387680/106095502-e7fe5b00-6101-11eb-85ca-86c5136a6897.mp4

Relevant frames:

| &nbsp; | &nbsp; |
| --- | --- |
| ![cascade-0003](https://user-images.githubusercontent.com/2387680/106095772-5ba06800-6102-11eb-8037-abfcd702d609.png) | ![cascade-0005](https://user-images.githubusercontent.com/2387680/106095774-5c38fe80-6102-11eb-8139-a41136375bcc.png) |
| ![cascade-0006](https://user-images.githubusercontent.com/2387680/106095775-5cd19500-6102-11eb-8159-211152420d0a.png) | ![cascade-0007](https://user-images.githubusercontent.com/2387680/106095776-5cd19500-6102-11eb-8c65-b76154fe2cd4.png) |

Here's another video captured at 5x (by slowing down animations in debugger settings):

https://user-images.githubusercontent.com/2387680/106095561-fe0c1b80-6101-11eb-9d35-c9aa2de738f2.mp4

**Comment by saket** [comment]
I'm watching a video of your app sent by @burntcookie90 and I think I know what's happening for your popup -- AppCompat's default background drawable for popups has internal paddings. They cause a gap to be present between your popup's last item and the popup's bottom edge. I'm fairly sure it's also causing the arrow to be visible for a frame when the submenu has fully entered in.

Can you try setting a custom background to your popup without internal paddings? 

```kotlin
val cascadeStyler = CascadePopupMenu.Styler(
  background = {
    PaintDrawable(popupColor).also { it.setCornerRadius(8f.dip) }
  }
)
val cascade = CascadePopupMenu(context, anchorView, styler = cascadeStyler)
```

I wrongly assumed that https://github.com/saket/cascade/commit/79ace480f33bc7cd2726db5022c099deeb54d9c4 would be sufficient in working around this issue, but looks like I need to also add a default background for cascade popups that doesn't have internal paddings.

**Comment by saket** [comment]
Update: This should be fixed by https://github.com/saket/cascade/commit/667ffd242a5992cd1b9187818a11cbd64bba1aab. Can you try out `1.3.0-SNAPSHOT`?

**Comment by cmt218** [comment]
Confirmed fixed in `1.3.0` thanks for the help!
